hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
Actually [`Cluster.getClusterVersion` should return `Version.UNKNOWN`](https://github.com/vbekiaris/hazelcast/blob/c08968bd5592d56fcd5590342cc0cafcfe3bf7d0/hazelcast/src/main/java/com/hazelcast/core/Cluster.java#L193-L195) until node joins, so I think the suggestion to send out the password does make sense --> upgrade from 3.8 & 3.8.1 should be still possible.